packages/teams-js/test/public/authentication.spec.ts

```diff
@@ -461,431 +499,857 @@ describe('Testing authentication module', () => {
-          expect(message).not.toBeNull();
-          expect(message.args.length).toBe(1);
-          expect(message.args[0]).toBe(mockResult);
-        });
-      } else {
-        it(`authentication.notifySuccess should not allow calls from ${context} context`, async () => {
-          await utils.initializeWithContext(context);
-          expect(() => authentication.notifySuccess()).toThrow(
-            `This call is only allowed in following contexts: ${JSON.stringify(
-              allowedContexts,
-            )}. Current context: "${context}".`,
-          );
-        });
-      }
+            authentication.notifyFailure(
+              '',
+              'https%3A%2F%2Fsomeinvalidurl.com%3FcallbackUrl%3Dtest%23%2Fconfiguration',
+            );
+            const message = utils.findMessageByFunc('authentication.authenticate.failure');
+            expect(message).not.toBeNull();
+            expect(message.args.length).toBe(1);
+            expect(message.args[0]).toBe('');
+          });
+
+          it(`authentication.notifyFailure should successfully notify auth failure if callbackUrl and reason are empty from ${context} context`, async () => {
+            await utils.initializeWithContext(context);
+
+            authentication.notifyFailure('', '');
+            const message = utils.findMessageByFunc('authentication.authenticate.failure');
+            expect(message).not.toBeNull();
+            expect(message.args.length).toBe(1);
+            expect(message.args[0]).toBe('');
+          });
+
+          it(`authentication.notifyFailure should successfully notify auth failure if callbackUrl is empty from ${context} context`, async () => {
+            await utils.initializeWithContext(context);
+
+            authentication.notifyFailure(errorMessage, '');
+            const message = utils.findMessageByFunc('authentication.authenticate.failure');
+            expect(message).not.toBeNull();
+            expect(message.args.length).toBe(1);
+            expect(message.args[0]).toBe(errorMessage);
+          });
+        } else {
+          it(`authentication.notifyFailure should not allow calls from ${context} context`, async () => {
+            await utils.initializeWithContext(context);
+            expect(() => authentication.notifyFailure()).toThrow(
+              `This call is only allowed in following contexts: ${JSON.stringify(
+                allowedContexts,
+              )}. Current context: "${context}".`,
+            );
+          });
+        }
+      });
     });
   });
 
-  describe('Testing authentication.notifyFailure', () => {
-    const allowedContexts = [FrameContexts.authentication];
-    it('authentication.notifyFailure should not allow calls before initialization', () => {
-      expect(() => authentication.notifyFailure()).toThrowError('The library has not yet been initialized');
+  describe('FRAMELESS - authentication tests', () => {
```

Comment:
Rather than duplicate all of these tests, should we just be looping through each one twice and providing it the framed and frameless mocks (kind of like how we iterate through framecontexts in tests in other places)?

Reply:
I have opened a bug to track investigating this approach